### PR TITLE
Flashlight improvements and fixes, also fixed potential VRAM leak in ctc_models.transcribe

### PR DESCRIPTION
**Pull request by trias702**
# What does this PR do ?

Flashlight improvements and fixes, also fixed potential VRAM leak in ctc_models.transcribe

**Collection**: ASR

# Changelog 
- Set logits to CPU in ctc_models.transcribe
- Changed all `self.decoder.vocabulary` calls in ctc_models to be `OmegaConf.to_container(self.decoder.vocabulary)`, this is necessary for performance reasons, because downstream decoders like Flashlight access the vocabulary a lot, and if it's passed as a ListConfig, it's incredibly expensive to access repeatedly, hence pass as Python list instead
- Made changes to silence/unknown token handling in flashlight decoder
- Removed the unit_lm boolean requirement in flashlight_cfg. It's not necessary, the decoder knows that if you pass a lexicon it should use it, otherwise do lexicon-free mode. There's no need for a boolean too, it's superfluous, the presence of the lexicon file is the only indicator the class needs

# Usage
No changes to how anything is used, except don't pass `unit_lm` anymore to `flashlight_cfg`

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [X] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/models/ctc_models.py [hunk 1]
```diff
@@ -196,6 +196,7 @@ def transcribe(
                     logits, logits_len, greedy_predictions = self.forward(
                         input_signal=test_batch[0].to(device), input_signal_length=test_batch[1].to(device)
                     )
+                    logits = logits.cpu()
```
Comment (titu1994) [inline nemo/collections/asr/models/ctc_models.py:199]: Can you add this to hybrid transcribe as well (CTC part)? And maybe other models that have transcribe implementation can benefit from this

**Review by titu1994 — COMMENTED** [review]
Looks good to me, thanks for the improvements ! One thing is can you check if other transcribe calls like hybrid are also leaking this memory

**Comment by trias702** [comment]
Absolutely, happy to help! I'll look into the hybrid and RNNT models first
thing next week.

File: nemo/collections/asr/models/ctc_models.py (hunk 1 above)
Comment (trias702) [reply nemo/collections/asr/models/ctc_models.py]: Sounds good, I have checked and fixed the same VRAM leak in hybrid and classification models too. I've also made the fix better for ctc_models so the cast back to cpu happens after decoding has run (so decoding can still gain benefit of GPU speed).

**Review by titu1994 — APPROVED** [review]
Looks good to me, thanks !
